File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
         if(x<0){
             return false;
         }

         long long  org=x;
         long long rev =0;
         while(x>0){
            long dig=x%10;
            rev=rev*10+dig;
            x=x/10;
         }

       return  org == rev; 
        
    }
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        # Negative numbers are not palindromes
        if x < 0:
            return False
        
        # Convert the number to a string
        s = str(x)
        
        # Check if the string reads the same backward and forward
        return s == s[::-1]
